### PR TITLE
remove unnecessary std::string in source

### PR DESCRIPTION
**Pull request by ME-ON1**
Signed-off-by: Tarun Sharma <starun.1998@gmail.com>

https://github.com/envoyproxy/envoy/issues/11318

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/config/grpc_mux_impl.cc [hunk 1]
```diff
@@ -316,7 +316,7 @@ void GrpcMuxImpl::queueDiscoveryRequest(absl::string_view queue_item) {
     api_state.pending_ = true;
     return; // Drop this request; the unpause will enqueue a new one.
   }
-  request_queue_->emplace(std::string(queue_item));
+  request_queue_->emplace(absl::string_view(queue_item));
```
Comment (yanavlasov) [inline source/common/config/grpc_mux_impl.cc:319]: Since `queue_item` is already `absl::string_view` type, this cast is not necessary. Same above.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
/wait

File: source/common/config/grpc_mux_impl.cc (hunk 1 above)
Comment (ME-ON1) [reply source/common/config/grpc_mux_impl.cc:319]: let me change it then.

**Comment by ME-ON1** [comment]
@yanavlasov  can you take a another look i have updated this PR and will also add few more if this works. 

Also the test which are failing in azure CI is passed locally, do we have to run it again ?

**Review by yanavlasov — APPROVED** [review]

File: source/common/common/utility.cc [hunk 2]
```diff
@@ -192,9 +192,9 @@ void DateFormatter::parse(const std::string& format_string) {
   }
 }
 
-std::string
-DateFormatter::fromTimeAndPrepareSpecifierOffsets(time_t time, SpecifierOffsets& specifier_offsets,
-                                                  const std::string& seconds_str) const {
+std::string DateFormatter::fromTimeAndPrepareSpecifierOffsets(time_t time,
+                                                              SpecifierOffsets& specifier_offsets,
+                                                              absl::string_view seconds_str) const {
```
Comment (yanavlasov) [inline source/common/common/utility.cc:197]: To preserve original constness, please consider.

```suggestion
                                                              const absl::string_view seconds_str) const {
```

File: source/common/common/utility.cc [hunk 3]
```diff
@@ -435,7 +435,7 @@ std::string StringUtil::subspan(absl::string_view source, size_t start, size_t e
   return std::string(source.data() + start, end - start);
 }
 
-std::string StringUtil::escape(const std::string& source) {
+std::string StringUtil::escape(absl::string_view source) {
```
Comment (yanavlasov) [inline source/common/common/utility.cc:438]: To preserve original constness, please consider.
```suggestion
std::string StringUtil::escape(const absl::string_view source) {
```

**Review by yanavlasov — CHANGES_REQUESTED** [review]
You may also want to merge main to pick up fixes for a flaky test.

File: source/common/common/utility.cc (hunk 2 above)
Comment (ME-ON1) [reply source/common/common/utility.cc:197]: sure

File: source/common/common/utility.cc (hunk 3 above)
Comment (ME-ON1) [reply source/common/common/utility.cc:438]: sure

**Review by rojkov — APPROVED** [review]
Thank you! Looks good to me.
Deferring to @yanavlasov for a final pass and merge.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**